### PR TITLE
DT-654 return consistent Legal Status

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/net/syscon/elite/api/resource/BookingResource.java [hunk 1]
```diff
@@ -590,6 +590,7 @@ InmateDetail getOffenderBookingByOffenderNo(@ApiParam(value = "The offenderNo of
     ResponseEntity<byte[]> getMainBookingImageDataByNo(@ApiParam(value = "The offender No of offender", required = true) @PathVariable("offenderNo") String offenderNo,
                                                   @ApiParam(value = "Return full size image", defaultValue = "false") @RequestParam(value = "fullSizeImage", defaultValue = "false", required = false) boolean fullSizeImage);
 
+    @Deprecated
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/resource/BookingResource.java:593]: This API is not used to marking deprecated

File: src/main/java/net/syscon/elite/api/resource/impl/UserResourceImpl.java [hunk 2]
```diff
@@ -198,23 +211,14 @@ public UserDetail getUserDetails(final String username) {
 
     @Override
     public ResponseEntity<List<OffenderBooking>> getMyAssignments(final Long pageOffset, final Long pageLimit) {
-        final var nomisProfile = ProfileUtil.isNomisProfile(env);
         var iepLevel = false;
-        List<Long> bookingIds = null;
+        List<Long> bookingIds = inmateService.getPersonalOfficerBookings(authenticationFacade.getCurrentUsername());
         List<String> offenderNos = null;
 
-        if (nomisProfile) {
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/resource/impl/UserResourceImpl.java:206]: Personal officers are not needed in Nomis - only in Syscon elite - we use keyworkers now in different service

File: src/main/java/net/syscon/elite/repository/impl/InmateRepositoryImpl.java [hunk 3]
```diff
@@ -81,7 +82,6 @@
             .put("FACE_IMAGE_ID", new FieldMapper("facialImageId"))
             .put("LIVING_UNIT_ID", new FieldMapper("assignedLivingUnitId"))
             .put("LIVING_UNIT_DESC", new FieldMapper("assignedLivingUnitDesc", value -> RegExUtils.replaceFirst((String) value, "^[A-Z|a-z|0-9]+\\-", "")))
-            .put("ASSIGNED_OFFICER_ID", new FieldMapper("assignedOfficerId"))
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/repository/impl/InmateRepositoryImpl.java:84]: Field is redundant

File: src/main/java/net/syscon/elite/api/model/InmateDetail.java [hunk 4]
```diff
@@ -131,9 +132,6 @@
     @ApiModelProperty(value = "Category code (from list of assessments)")
     private String categoryCode;
 
-    @ApiModelProperty(value = "Staff Id reference for assigned officer / keyworker")
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/model/InmateDetail.java:134]: This attribute is never used

Comment (Mjwillis) [reply src/main/java/net/syscon/elite/api/model/InmateDetail.java:134]: Its also misleading

File: src/main/java/net/syscon/elite/api/model/InmateDetail.java [hunk 5]
```diff
@@ -161,8 +159,8 @@
     @ApiModelProperty(value = "Status of prisoner", required = true, example = "ACTIVE IN", allowableValues = "ACTIVE IN,ACTIVE OUT", position = 18)
     private String status;
 
-    @ApiModelProperty(value = "Legal Status", example = "Convicted", allowableValues = "Convicted,Remand", notes = "Only returned when requesting extra details")
-    private String legalStatus;
+    @ApiModelProperty(value = "Legal Status", example = "REMAND", notes = "Only returned when requesting extra details")
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/model/InmateDetail.java:162]: Only prisoner offender search uses this value at the moment.

File: src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        if (mainLegalStatusCode.equals("UNKNOWN")) {
+            return LegalStatus.UNKNOWN;
+        }
+
+        return LegalStatus.OTHER;
+    }
+
+    public static String getConvictedStatus(final String bandCode) {
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java:79]: Kept backward compatibility with old logic for now

File: src/main/resources/db/migration/data/R__3_6_1__OFFENDER_BOOKINGS.sql [hunk 7]
```diff
@@ -59,7 +59,7 @@ INSERT INTO OFFENDER_BOOKINGS (OFFENDER_BOOK_ID, BOOKING_BEGIN_DATE, BOOKING_NO,
         (-48, TO_DATE('2000-08-16', 'YYYY-MM-DD'), 'A09113', -1006, 2, 'N', 'OUT',  'N', 'N', 'OUT', -1006, null),
 
 -- For adjudications.feature
-        (-49, TO_DATE('2000-08-16', 'YYYY-MM-DD'), 'Z00045', -1048, 2, 'N', 'LEI',  'N', 'N', 'LEI', -1048, null),
+        (-49, TO_DATE('2000-08-16', 'YYYY-MM-DD'), 'Z00045', -1048, 2, 'N', 'OUT',  'N', 'N', 'OUT', -1048, null),
```
Comment (Mjwillis) [inline src/main/resources/db/migration/data/R__3_6_1__OFFENDER_BOOKINGS.sql:62]: This data is wrong, you can't have an IN OUT status of LEI

File: src/main/java/net/syscon/elite/api/model/PrisonerInformation.java [hunk 8]
```diff
@@ -75,15 +81,8 @@
...
-    private String legalStatus;
-
-    public void deriveLegalStatus(String bandCodeStr) {
-        if (bandCodeStr != null) {
-            final var bandCode = Integer.parseInt(bandCodeStr);
-            legalStatus = (bandCode <= 8 || bandCode == 11) ? "Convicted" : "Remand";
-        }
-    }
+    @ApiModelProperty(value = "Legal Status", example = "REMAND", position = 19)
+    private LegalStatus legalStatus;
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/model/PrisonerInformation.java:85]: This entity is only used by NHS service

Comment (Mjwillis) [reply src/main/java/net/syscon/elite/api/model/PrisonerInformation.java:85]: Which is not live yet

File: src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+            this.desc = desc;
+        }
+    }
+    
+    private final static List<String> RECALL_STATUS_CODES = List.of("14FTR_ORA","14FTRHDC_ORA","CUR_ORA","FTR/08","FTR_HDC","FTR_HDC_ORA","FTR_ORA","FTR_SCH15","FTRSCH15_ORA","HDR_ORA","LR","LR_ALP","LR_ALP_LASPO","LR_DLP","LR_DPP","LR_EPP","LR_ES","LR_HDC","LR_IPP","LR_LASPO_AR","LR_LASPO_DR","LR_LIFE","LR_MLP","LR_ORA","LR_SEC236A","LR_SEC91_ORA","LR_YOI","LR_YOI_ORA"); 
+
+    public static LegalStatus getLegalStatus(final String bandCode, final String mainLegalStatusCode) {
+
+        if (StringUtils.isBlank(bandCode) || !StringUtils.isNumeric(bandCode)) {
+            return null;
```
Comment (andymarke) [inline src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java:35]: Why not "unknown"?

File: src/main/java/net/syscon/elite/api/resource/UserResource.java [hunk 10]
```diff
@@ -100,17 +119,6 @@
             @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
     UserDetail getMyUserInformation();
 
-    @GetMapping("/me/bookingAssignments")
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/api/resource/UserResource.java:103]: Not used and incorrect info

File: src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+        if (RECALL_STATUS_CODES.contains(mainLegalStatusCode)) {
+            return LegalStatus.RECALL;
+        }
+
+        if (legalStatusBand == 0) {
+            return LegalStatus.DEAD;
+        }
+
+        if (legalStatusBand == 1) {
+            return LegalStatus.INDETERMINATE_SENTENCE;
```
Comment (andymarke) [inline src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java:49]: As a matter of interest does this included Life sentences?

File: src/main/java/net/syscon/elite/api/model/PrisonerInformation.java (hunk 8 above)
Comment (andymarke) [inline src/main/java/net/syscon/elite/api/model/PrisonerInformation.java:85]: Where does this get set?

File: src/main/java/net/syscon/elite/api/resource/UserResource.java (hunk 10 above)
Comment (andymarke) [inline src/main/java/net/syscon/elite/api/resource/UserResource.java:103]: I'm sure there is a good reason

File: src/test/resources/net/syscon/elite/api/resource/impl/offender_detail_aliases.json [hunk 12]
```diff
@@ -131,7 +130,7 @@
     }
   ],
   "status": "ACTIVE IN",
-  "legalStatus": "Convicted",
+  "legalStatus": "SENTENCED",
```
Comment (andymarke) [inline src/test/resources/net/syscon/elite/api/resource/impl/offender_detail_aliases.json:133]: So to confirm, this only affects offender search?

File: src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java (hunk 9 above)
Comment (Mjwillis) [reply src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java:35]: Yes was toying with that. - That could be in the next PR when I tighten up the Recall calc

File: src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java (hunk 11 above)
Comment (Mjwillis) [reply src/main/java/net/syscon/elite/api/model/LegalStatusCalc.java:49]: Don't know, I'll check

File: src/main/java/net/syscon/elite/api/resource/UserResource.java (hunk 10 above)
Comment (Mjwillis) [reply src/main/java/net/syscon/elite/api/resource/UserResource.java:103]: Not used any more

File: src/test/resources/net/syscon/elite/api/resource/impl/offender_detail_aliases.json (hunk 12 above)
Comment (Mjwillis) [reply src/test/resources/net/syscon/elite/api/resource/impl/offender_detail_aliases.json:133]: Yes

**Pull request merged**